sdk/python/feast/online_response.py

```diff
@@ -94,25 +95,30 @@ def _infer_online_entity_rows(
     entity_rows_dicts = cast(List[Dict[str, Any]], entity_rows)
     entity_row_list = []
     entity_type_map = dict()
+    entity_value_map = defaultdict(list)
+
+    # Flatten keys-value dicts into lists for type inference
+    for entity in entity_rows_dicts:
+        for key, value in entity.items():
+            if not isinstance(value, Value):
+                entity_value_map[key].append(value)
+
+    # Loop over all entities to infer dtype first in case of empty lists or nulls
+    for key, values in entity_value_map.items():
+        entity_type_map[key] = python_values_to_feast_value_type(key, values)
 
     for entity in entity_rows_dicts:
         fields = {}
         for key, value in entity.items():
-            # Allow for feast.types.Value
+            if key not in entity_type_map:
```

Comment:
this would break if the user added Value items in the entity rows right?

Reply:
I think I've addressed this now - let me know what you think.

Reply:
I think I've addressed this now - let me know what you think.

Reply:
Not sure if I'm missing something, but I think it's still an issue?

If the user passes in as entity rows something like

`[{"driver_id:" : Value}]`, then this would result in an empty entity_type_map and fail, when before it would have gone to the 

```
if isinstance(value, Value):
    proto_value = value
```
line

Reply:
ah sorry was looking at the wrong commit diff.  this does seem like it'd fix it.

Reply:
That feels like it's a different PR? I haven't actually changed any behaviour here and it wasn't tested before. Not saying it doesn't need a test just not sure this PR should keep growing and growing.

Reply:
sg. we can leave this for later